### PR TITLE
Restructure oneDPL specification 

### PR DESCRIPTION
**Pull request by ValentinaKats**
Restructure of oneDPL specification

### CODE REVIEW

**Comment by ValentinaKats** [comment]
@akukanov, @timmiesmith, could you please take a look at these changes?

File: source/elements/oneDPL/source/common.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
-.. SPDX-FileCopyrightText: 2019-2020 Intel Corporation
+.. SPDX-FileCopyrightText: 2019-2022 Intel Corporation
 ..
 .. SPDX-License-Identifier: CC-BY-4.0
 
 Namespaces
 ----------
 
-oneDPL uses :code:`namespace oneapi::dpl` for all its functionality including Parallel STL algorithms,
+oneDPL uses :code:`namespace oneapi::dpl` for all its functionality including Parallel algorithms,
```
Comment (akukanov) [inline source/elements/oneDPL/source/common.rst:8]:
```suggestion
oneDPL uses :code:`namespace oneapi::dpl` for all its functionality including parallel algorithms,
```

File: source/elements/oneDPL/source/index.rst [hunk 2]
```diff
@@ -15,23 +15,28 @@ implementing data parallel algorithms.
...
-- Extensions: an additional set of library classes and functions that
-  are known to be useful in practice but are not yet included into
-  C++ or SYCL specifications. (See :doc:`Specific API of oneDPL <extensions>`.)
+  * An additional set of library classes and functions that
+    are known to be useful in practice but are not yet included into
+    C++ or SYCL specifications. 
 
+- :doc:`SYCL Kernels API <sycl_kernels_api>`:
+
+  * A subset of the `C++ standard`_ library which you can use with
```
Comment (akukanov) [inline source/elements/oneDPL/source/index.rst:29]:
```suggestion
  * A subset of the `C++ standard`_ library which can be used with
```

The reason is that the spec is a document primarily for implementers, not for users

File: source/elements/oneDPL/source/parallel_api.rst [hunk 3]
```diff
@@ -1,36 +1,374 @@
...
+.. code:: cpp
+
+  template <typename NewKernelName, typename OldKernelName>
+  device_policy<NewKernelName>
+  make_device_policy( const device_policy<OldKernelName>& policy = dpcpp_default )
+
+Return a policy object constructed from ``policy``, with a new kernel name provided as the template
+argument. If no policy object is provided, the new policy is constructed from ``dpcpp_default``.
+
+Parallel Algorithms
```
Comment (akukanov) [inline source/elements/oneDPL/source/parallel_api.rst:142]: For now, let's keep "Parallel Algorithms" at its current place on the page, to minimize the amount of changes and simplify the review.

File: source/elements/oneDPL/source/sycl_kernels_api.rst [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+++++++++++++++++++++++++++++++++++++++++++++++++++
+
+oneDPL defines a subset of the C++ standard library APIs for use in DPC++ kernels. These APIs
+can be employed in the kernels similarly to how they are employed in code for a typical
+CPU-based platform.
+
+For all C++ algorithms accepting execution policies (as defined by `C++ Standard`_), oneDPL provides
+an implementation for oneAPI devices via :code:`oneapi::dpl::execution::device_policy`. These algorithms
+must be capable of processing data in SYCL buffers (passed via :code:`oneapi::dpl::begin/end`)
+and in unified shared memory (USM). (See :doc:`Parallel API <parallel_api>`.)
```
Comment (akukanov) [inline source/elements/oneDPL/source/sycl_kernels_api.rst:18]: Let's move this block to the beginning of the Parallel API doc.

File: source/elements/oneDPL/source/sycl_kernels_api.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+SYCL Kernels API
+----------------
+
+Supported C++ Standard Library APIs and Algorithms
+++++++++++++++++++++++++++++++++++++++++++++++++++
+
+oneDPL defines a subset of the C++ standard library APIs for use in DPC++ kernels. These APIs
```
Comment (akukanov) [inline source/elements/oneDPL/source/sycl_kernels_api.rst:11]:
```suggestion
oneDPL defines a subset of the `C++ standard`_ library APIs for use in DPC++ kernels. These APIs
```

File: source/elements/oneDPL/source/common.rst (hunk 1 above)
Comment (ValentinaKats) [reply source/elements/oneDPL/source/common.rst]: Applied suggestion

File: source/elements/oneDPL/source/index.rst (hunk 2 above)
Comment (ValentinaKats) [reply source/elements/oneDPL/source/index.rst]: Applied suggestion

File: source/elements/oneDPL/source/parallel_api.rst (hunk 3 above)
Comment (ValentinaKats) [reply source/elements/oneDPL/source/parallel_api.rst]: Applied suggestion

File: source/elements/oneDPL/source/sycl_kernels_api.rst (hunk 4 above)
Comment (ValentinaKats) [reply source/elements/oneDPL/source/sycl_kernels_api.rst]: Applied suggestion

File: source/elements/oneDPL/source/sycl_kernels_api.rst (hunk 5 above)
Comment (ValentinaKats) [reply source/elements/oneDPL/source/sycl_kernels_api.rst]: Applied suggestion

**Review by akukanov — APPROVED** [review]

**Pull request merged**